core/src/test/java/io/questdb/std/DirectByteCharSequenceIntHashMapTest.java

```diff
@@ -138,9 +141,39 @@ public void testHashMapCompatibility() {
                 }
                 p += len;
             }
+        } finally {
+            Unsafe.free(mem, memSize, MemoryTag.NATIVE_DEFAULT);
+        }
+    }
+
+    @Test
+    public void testHashMapUtf8() {
+        final int memSize = 64;
+        long mem = Unsafe.malloc(memSize, MemoryTag.NATIVE_DEFAULT);
+        final DirectByteCharSequence dbcs = new DirectByteCharSequence();
+        DirectByteCharSequenceIntHashMap map = new DirectByteCharSequenceIntHashMap();
+        try {
+            final String utf16Str = "таблица-табличка";
+            byte[] utf8Bytes = utf16Str.getBytes(StandardCharsets.UTF_8);
+            for (int i = 0; i < utf8Bytes.length; i++) {
+                Unsafe.getUnsafe().putByte(mem + i, utf8Bytes[i]);
+            }
+
+            dbcs.of(mem, mem + utf8Bytes.length);
+
+            Assert.assertEquals(-1, map.get(dbcs));
+
+            final StringSink sink = new StringSink();
+            final String utf8Str = sink.put(dbcs).toString();
+            map.put(utf8Str, 42);
+            Assert.assertEquals(42, map.get(dbcs));
+            Assert.assertEquals(42, map.get(utf8Str));
 
+            Assert.assertEquals(1, map.size());
```

Comment:
Can you also put by `dbcs` and get by `utf8Str`? Can you find and test hash collisions as well?

Reply:
> put by `dbcs`

This is not supported by these maps. We only put strings into them in the production code, so adding such overload would be synthetic and test-only.

Reply:
> Can you find and test hash collisions as well?

Not sure if I follow this question. Hash collisions should be analyzed against the hash function, not the end hash table. [This repo](https://github.com/rurban/smhasher) has results for xxHash64 which is the hash function we use. Am I missing something?

Reply:
> Perhaps it's indirectly tested in ILP tests?

Yes, that's the case, but adding more cases here is a good idea. Let me do that.

Reply:
Done in efbac9600c69e97078e6c73e7407d3515c9deeb3